controllers/drcluster_controller.go

```diff
@@ -175,7 +177,10 @@ func drpcPred() predicate.Funcs {
 			return false
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			return false
+			o := e.Object
+			log.Info("Processing DRPC delete event", "name", o.GetName(), "namespace", o.GetNamespace())
+
+			return true
```

Comment:
I'm not sure about this, does delete actually have the resource that was deleted? The further calls here would go to `filterDRPC` and that would return the failoverCluster for reconciliation, assuming we have the object which on reading does not necessarily imply so.

I would ignore the delete in the watcher, or unconditionally trigger a reconcile for all DRClusters on a DRPC deletion detection.

Currently a stale MMode would be left behind if a failover was initiated, that caused an MMode to be set, and subsequently the DRPC was deleted before the failover was complete. We can note this for now, and not fix it as such.

Reply:
We are doing exactly the same thing [here](https://github.com/RamenDR/ramen/blob/cae74b9e3eb2c742143a828767df9d6c7406798e/controllers/volumereplicationgroup_controller.go#L196), and then we are also filtering [here](https://github.com/RamenDR/ramen/blob/cae74b9e3eb2c742143a828767df9d6c7406798e/controllers/volumereplicationgroup_controller.go#L117-L129), so I assumed that we are actually having the resource, that was deleted

Reply:
I am still not sure this helps us trap delete events of interest, we have the object type latch in `drpcMapFun` so should not cause panics or errors.